controllers/nstemplateset/cluster_resources.go

```diff
@@ -35,14 +40,19 @@ type listExistingResources func(cl runtimeclient.Client, username string) ([]run
 type toolchainObjectKind struct {
 	gvk                   schema.GroupVersionKind
 	object                runtimeclient.Object
-	listExistingResources listExistingResources
+	listExistingResources listExistingResourcesIfAvailable
 }
 
 func newToolchainObjectKind(gvk schema.GroupVersionKind, emptyObject runtimeclient.Object, listExistingResources listExistingResources) toolchainObjectKind {
 	return toolchainObjectKind{
-		gvk:                   gvk,
-		object:                emptyObject,
-		listExistingResources: listExistingResources,
+		gvk:    gvk,
+		object: emptyObject,
+		listExistingResources: func(cl runtimeclient.Client, username string, availableAPIGroups []metav1.APIGroup) (objects []runtimeclient.Object, e error) {
```

Comment:
there's something a bit confusing here: `listExistingResources` has the same type as `listExistingResourcesIfAvailable` at line 33 above, but it uses the same name as type in line 30.

maybe we should just drop the `listExistingResourcesIfAvailable` type definition, if it's not used anywhere else?

Reply:
good point - didn't notice that cb30d01